src/any_agent/frameworks/langchain.py

```diff
@@ -65,28 +66,69 @@ async def _load_agent(self) -> None:
         for mcp_server in mcp_servers:
             imported_tools.extend(mcp_server.tools)
 
-        model = self._get_model(self.config)
+        if self.managed_agents:
+            swarm = []
+            managed_names = []
+            for n, managed_agent in enumerate(self.managed_agents):
+                managed_tools, managed_mcp_servers = await import_and_wrap_tools(
+                    managed_agent.tools, agent_framework=AgentFramework.LANGCHAIN
+                )
+                managed_tools.extend(
+                    [
+                        tool
+                        for mcp_server in managed_mcp_servers
+                        for tool in mcp_server.tools
+                    ]
+                )
+                name = managed_agent.name
+                if not name or name == "any_agent":
+                    logger.warning(
+                        "Overriding name for managed_agent. Can't use the default."
+                    )
+                    name = f"managed_agent_{n}"
+                managed_names.append(name)
+
+                managed_agent = create_react_agent(
+                    name=name,
+                    model=self._get_model(managed_agent),
+                    tools=managed_tools
+                    + [create_handoff_tool(agent_name=self.config.name)],
+                    prompt=get_instructions(managed_agent.instructions),
+                    **managed_agent.agent_args or {},
+                )
+                swarm.append(managed_agent)
+
+            imported_tools += [
+                create_handoff_tool(agent_name=managed_name)
+                for managed_name in managed_names
+            ]
 
-        self._agent: CompiledGraph = create_react_agent(
-            model=model,
-            tools=imported_tools,
-            prompt=self.config.instructions,
-            **self.config.agent_args or {},
-        )
-        # Langgraph doesn't let you easily access what tools are loaded from the CompiledGraph, so we'll store a list of them in this class
-        self._tools = imported_tools
+            main_agent = create_react_agent(
+                name=self.config.name,
+                model=self._get_model(self.config),
+                tools=imported_tools,
+                prompt=get_instructions(self.config.instructions),
```

Comment:
The implementation in google AnyAgent is 
`get_instructions(managed_agent.instructions) or "",`

not sure if one style should be used everywhere

Reply:
nah, is just a quirk in the google API where they use an empty string as default argument but don't support `None` 🤷